loader/loader.c

```diff
@@ -6032,7 +6062,8 @@ VkResult setup_loader_term_phys_devs(struct loader_instance *inst) {
             goto out;
         }
         inst->total_gpu_count += icd_phys_dev_array[icd_idx].count;
-        icd_phys_dev_array[icd_idx].this_icd_term = icd_term;
+        icd_phys_dev_array[icd_idx++].this_icd_term = icd_term;
+        icd_term = icd_term->next;
```

Comment:
Ack, not a fan of the `icd_idx++` inside the indexing operator just cause it makes it not super clear which index will be used. I know its post increment, but any code that gets added after this would have to know to remove the `++` here and add it below.